src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -410,36 +424,14 @@ export class ObjectExplorerService implements IObjectExplorerService {
 						allProviders.push(...nodeProviders);
 					}
 
-					self._sessions[session.sessionId!].nodes[node.nodePath].expandEmitter.event((expandResult: NodeExpandInfoWithProviderId) => {
-						if (expandResult && expandResult.providerId) {
-							resultMap.set(expandResult.providerId, expandResult);
-							// If we got an error result back then send error our error event
-							// We only do this for the MSSQL provider
-							if (expandResult.errorMessage && expandResult.providerId === mssqlProviderName) {
-								const errorType = expandResult.errorMessage.indexOf('Object Explorer task didn\'t complete') !== -1 ? 'Timeout' : 'Other';
-								// For folders send the actual name of the folder (since the nodeTypeId isn't useful in this case and the names are controlled by us)
-								const nodeType = node.nodeTypeId === NodeType.Folder ? node.label : node.nodeTypeId;
-								this._telemetryService.createErrorEvent(TelemetryKeys.TelemetryView.Shell, TelemetryKeys.TelemetryError.ObjectExplorerExpandError, undefined, errorType)
-									.withAdditionalProperties({
-										nodeType,
-										providerId: expandResult.providerId
-									}).send();
-							}
-
-						} else {
-							this.logService.error('OE provider returns empty result or providerId');
-						}
+					self._sessions[session.sessionId!].nodes[node.nodePath].expandEmitter.event(
+						(expandResult: NodeExpandInfoWithProviderId) =>
+							resolve(this.completedExpandRequest(expandResult, resultMap, node, newRequest, session, allProviders)));
```

Comment:
Now you have a new problem - you resolve this immediately on the first provider that you get results from. But that's almost positively going to be undefined with how completedExpandRequest is structured.

The original code was in a promise because it needed to collect responses from all providers before finishing. So you need to handle that here too.

Simplest would be to only call resolve if completedExpandRequest returns a non-undefined value (since it only returns that when the resultMap is full). Not the cleanest way of solving this though - but other ways would involve more rewriting.

Reply:
> The original code was in a promise because it needed to collect responses from all providers before finishing. So you need to handle that here too.

Didn't you say [above](https://github.com/microsoft/azuredatastudio/pull/21590#discussion_r1072676014):
> This doesn't need to create a new promise - there's no callbacks you have to call resolve from. And it also doesn't seem to need to return a Promise at all, I'm not seeing any async code in here. It can just return mergeResults directly.